wiki/608483-BuildaTool.md

```diff
@@ -0,0 +1,123 @@
+## Overview
+
+The NLP Sandbox aims to streamline the development and benchmarking of NLP Tools
+that are:
+
+- Robust and reliable
+- Reproducible
+- Re-usable
+- Portable
+- Cloud-friendly
+
+The sections below describe how to build and benchmark your first NLP Tool using
+the NLP Sandbox. But beforehand, let's briefly discuss some of the key aspects
+of the design of the NLP Sandbox.
+
+## API (micro-)services vs programs
+
+Most developers are familiar with the development of command-line programs.
+Examples of programs include Bash scripts, Python scripts, Java executables,
+etc. The behavior of a program is usually defined by command line arguments
+and/or the use of a configuration file. One point that characterizes a program
+is that after completing its initialization (e.g., detection of GPUs) and
+primary task (e.g., processing data), the program shuts down.
+
+Another characteristic of *using* a program - from a user perspective - is that
+unless you use a container technology like Docker or Singularity, running a
+Python/Java/other program requires that you install a Python/Java/other
+environment beforehand.
+
+An Application Programming Interface (API) service is different from a program
```

Comment:
Can potentially link out to existing resources that talk about APIs in more depth such as:

* https://www.ibm.com/cloud/learn/api
* https://en.wikipedia.org/wiki/API

Reply:
Done